pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -306,6 +306,13 @@ func (desc *wrapper) validateInboundTableRef(
 		// descriptor is for a sequence. In this case, they refer to the columns
 		// in the referenced descriptor instead.
 		for _, colID := range by.ColumnIDs {
+			// Skip this check if the column ID is zero. This can happen due to
+			// bugs in 20.2.
+			//
+			// TODO(ajwerner): Make sure that a migration in 22.2 fixes this issue.
```

Comment:
Can you file an issue to track this, please? We could start a 22.2 column in the project board.

Reply:
https://github.com/cockroachdb/cockroach/issues/83124